osquery/tables/system/windows/windows_eventlog.cpp

```diff
@@ -87,61 +85,91 @@ Status parseWelXml(QueryContext& context, std::wstring& xml_event, Row& row) {
 void renderQueryResults(QueryContext& context,
                         EVT_HANDLE queryResults,
                         RowYield& yield) {
-  std::vector<EVT_HANDLE> events(kNumEventsBlock);
-  unsigned long numEvents = 0;
-
-  // Retrieve the events one block at a time
-  auto ret = EvtNext(
-      queryResults, kNumEventsBlock, events.data(), INFINITE, 0, &numEvents);
-  while (ret != FALSE) {
-    for (unsigned long i = 0; i < numEvents; i++) {
-      unsigned long renderedBuffSize = 0;
-      unsigned long renderedBuffUsed = 0;
-      unsigned long propCount = 0;
-      if (!EvtRender(nullptr,
-                     events[i],
-                     EvtRenderEventXml,
-                     renderedBuffSize,
-                     nullptr,
-                     &renderedBuffUsed,
-                     &propCount)) {
-        if (GetLastError() != ERROR_INSUFFICIENT_BUFFER) {
-          LOG(WARNING) << "Failed to get the size of rendered event "
+  uint32_t kNumEventsBlock = 1024;
+  uint32_t _position = 0;
+
+  // The batch size should be more than 32. It is not documented
+  // but `EvtNext` should not fail (RPC_S_INVALID_BOUND error)
+  // with low batch size.
+  while (kNumEventsBlock > 32) {
+    std::vector<EVT_HANDLE> events(kNumEventsBlock);
```

Comment:
Do you want to reallocate this vector each time? I see that the size may change, but you could resize it instead since you will start at the largest possible size and potentially decrease.

I ask this because reallocating tends to be less performant compared to resizing.

Reply:
Thanks @theopolis for the review. I updated it to resize the vector if batch size is changed after invalid bound error.